src/interfaces.js

```diff
@@ -101,6 +102,31 @@ export const devErrorHandler = (err, req, res, next) => {
   })
 }
 
+/**
+ * Sets device infor available in req object for access in controllers
+ * @memberof module:device
+ * @method
+ * @param {http.ClientRequest} a native js http client request object
+ * @param {http.ServerResponse} a native js http server response object
+ * @param {function}
+ * @returns {function}
+ *
+ */
+export const deviceMiddleware = (req, res, next) => {
+  deviceIPLocation.getInfo(req.headers['user-agent'], req.ip, (err, res) => {
+    if (err !== undefined) {
+      devErrorHandler(err, req, res, next)
```

Comment:
There is no need to handle the error here, also no way of knowing the `env` context

Reply:
yes, linter will not allow

Reply:
maybe `if(!err)` ?

Reply:
@dmytro-kulchytskyi sure? I think Nico's point was more so handling the message entirely. linter does not like if the err var goes un noticed or handled.

Reply:
```
src/interfaces.js:116:63: Expected error to be handled.
```

Reply:
@amanelis yup, you right. So i think u can do something like that: 
`if (err) { 
next(err) 
}`

Reply:
We can't handle the error in the context of the `env` here because the package has no knowledge of the `env` the server is running on. Meaning it can't know its `devErrorHandler` as opposed to `errorHandler`. That job is for the injection container. 
Also, if there is an error here we don't want to stop the `req` since this is not a breaking exception. 
Just remove the error and add this at top of function:

`/* eslint-disable-next-line */`

Reply:
@dmytro-kulchytskyi perfect, that makes more sense.